### DISCUSSION TITLE
How to add new medium in Ncert tenant

### DISCUSSION THREAD

**Discussion by ArunachalamDhanapal** [Sunbird-Knowlg/Community#176] [General Discussions]
Hi team,
  Here i have to add new medium, gradLevel, subject values in Ncert tenant, I have script for that which is shared by ekstep implementation team, But that script is not working in Ncert production. So need help from your team to fix the issue.

**Comment by pallakartheekreddy** [comment]
@ArunachalamDhanapal Can you provide more details? What challenges are you encountering while executing the script?

**Comment by ArunachalamDhanapal** [comment]
@pallakartheekreddy For updating the framework, we pass xlsx sheet in the script, in that xlsx sheet, Board, medium, grade, and subject values could be passed, For other tenant BMGS values could working fine, but in Ncert there is no Board values, only Medium, Grade, Subject is available, So without Board value the script is not working. Can you provide solution for that without board, script will execute successfully.

Thank you.

**Reply by amitpriyadarshi** [reply]
@ArunachalamDhanapal in the same script, refer ntp.fwwoboard.FWMainFile (fwwoboard: Framework WithOut Board). This will help to create/update framework without board. Also refer line number 47 for excel sheet instruction. But you need to go through the script once in case of any issue. As from long time we also have not used this.
cc: @mohitga @pallakartheekreddy

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi I checked the script, It was already show some error related to cell type, But my doubt, in that script where we need to pass the data's, xlsx sheets and environment change, channel value and Id's on the tenant, Any instruction or sample video is available for run the script, i could'nt find anything. Can you help on this?.
 And my doubt is, This ( ntp.fwwoboard.FWMainFile ) script is userd for creating Medium grade subject in Ncert Tenant?

**Comment by amitpriyadarshi** [comment]
@ArunachalamDhanapal 
For Config File: refer line number: 98 of ntp.fwwoboard.FWMainFile.java
`IniFile loadConfig = new IniFile(strFilePath + "/config/config.ini");`

For Passing Excel Sheet: refer line number: 53-54 of ntp.fwwoboard.FWMainFile.java
`System.out.print("Enter Full Path of Excel file including name:: ");
			String xlsfile = br.readLine();`

For passing channel and framwork specific info: ref line number: 64-95 of ntp.fwwoboard.FWMainFile.java

`System.out.print("Enter Framework Name - Should be of format 'State (State Name)':: ");
			String strFrameworkName = br.readLine(); // Ex: 98e09d6e-b95b-4832-bfab-421e63d36aa7
			if (strFrameworkName == null || strFrameworkName.isEmpty()) {
				System.out.println("Framework Name is mandatory");
				System.exit(0);
			}

			System.out.print("Enter Framework Id:: ");
			String strFrameworkId = br.readLine();
			if (strFrameworkId == null || strFrameworkId.isEmpty()) {
				System.out.println("Framework Id is mandatory");
				System.exit(0);
			}

			String strTranslationCode = "";
			if (iTaxonomyUploadType == 2) {
				System.out.print("Enter Translation Code:: ");
				strTranslationCode = br.readLine();

				if (strTranslationCode == null || strTranslationCode.isEmpty()) {
					System.out.println("Translation Code is mandatory");
					System.exit(0);
				}

			}

			System.out.print("Enter Channel:: ");
			String strChannel = br.readLine();
			if (strChannel == null || strChannel.isEmpty()) {
				System.out.println("Channel is mandatory");
				System.exit(0);
			}`


Q. This ( ntp.fwwoboard.FWMainFile ) script is userd for creating Medium grade subject in Ncert Tenant?
Y. yes it is for medium, grade,subject.

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi Config, File path, channel and framework specific info we got it, what we pass in Translation Code, this is mandatory? We are try to run the script without Translation code, Script was running, but data do not reflect in Ncert tenant.

String strTranslationCode = "";
		if (iTaxonomyUploadType == 2) {
			System.out.print("Enter Translation Code:: ");
			strTranslationCode = br.readLine();

			if (strTranslationCode == null || strTranslationCode.isEmpty()) {
				System.out.println("Translation Code is mandatory");
				System.exit(0);
			}

		}

**Reply by amitpriyadarshi** [reply]
@ArunachalamDhanapal translationCode is not mandatory.  Also you need to publish the framework post creation. Just cross check the job execution log. All the api calls are successful or not?

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi Yes am publish the framework post creation, Will check the job execution logs for all apis.

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi I try to run the script, Script was running, but the values are not update, Can we connect a call, for screensharing session, It might be help full for us.

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi I run the script successfully, but my issue, when i login as state user, the data reflect in UI, without login as a state user,  the data are not reflect in UI. I attached a screenshot for your reference.
![UI_reflect_With login](https://github.com/Sunbird-Knowlg/Community/assets/122353629/dcee0b7f-816c-4eae-8f3b-519b6a7f91ee)
![UI_Not reflect_without login](https://github.com/Sunbird-Knowlg/Community/assets/122353629/e15e61f8-c072-4f2b-b680-b73521d535f5)

**Comment by ArunachalamDhanapal** [comment]
@amitpriyadarshi Please confirm, can we create a subject in the Punjabi language (ਵਿਸ਼ਾ) through scripts?
